ample/util/tests/test_options_processor.py

```diff
@@ -1,39 +1,90 @@
-    def test_process_mr_options(self):
-        """Make sure we handle resolution limits around SHELXE use properly
+
+    def test_options_shelxe_resolution_max(self):
+        """Make sure we turn off shelxe if resolution is outside limit
         """
         options = AMPLEConfigOptions()
-        argso = argparse_util.process_command_line(args=['-mtz', 'foo', '-fasta', 'bar'])
+        argso = argparse_util.process_command_line(args=['-mtz', 'foo',
+                                                         '-fasta', 'bar'])
         options.populate(argso)
 
-        # Make sure we turn off shelxe if resolution is outside limit
         self.assertTrue(options.d['use_shelxe'])  # default so should be true
         options.d['shelxe_rebuild'] = True
+
         # Set resolution below limit
         options.d['mtz_min_resolution'] = SHELXE_MAX_PERMITTED_RESOLUTION + 1
         options_processor.process_mr_options(options.d)
+
         self.assertFalse(options.d['use_shelxe'])
         self.assertFalse(options.d['shelxe_rebuild'])
 
-        # Check we don't accidently turn it off if above limit
-        options.d['use_shelxe'] = True
+
+    def test_options_shelxe_resolution_ok(self):
+        """Check we don't accidently turn it off if above limit
+        """
+        options = AMPLEConfigOptions()
+        argso = argparse_util.process_command_line(args=['-mtz', 'foo',
+                                                         '-fasta', 'bar'])
+        options.populate(argso)
+
+        self.assertTrue(options.d['use_shelxe'])  # default so should be true
         options.d['shelxe_rebuild'] = True
         options.d['mtz_min_resolution'] = SHELXE_MAX_PERMITTED_RESOLUTION - 1
         options_processor.process_mr_options(options.d)
+        
+        self.assertTrue(options.d['use_shelxe'])
+        self.assertTrue(options.d['shelxe_rebuild'])
+
+
+    def test_options_shelxe_resolution_cc(self):
+        """Check limit with coiled-coil mode turned on
+        """
+        options = AMPLEConfigOptions()
+        argso = argparse_util.process_command_line(args=['-mtz', 'foo',
+                                                         '-fasta', 'bar',
+                                                         '-coiled_coil', 'True'])
+        options.populate(argso)
+
+        options.d['shelxe_rebuild'] = True
+        limit = SHELXE_MAX_PERMITTED_RESOLUTION + ((SHELXE_MAX_PERMITTED_RESOLUTION_CC - SHELXE_MAX_PERMITTED_RESOLUTION) / 2)
+        options.d['mtz_min_resolution'] = limit
+        options_processor.process_mr_options(options.d)
+        
+        self.assertTrue(options.d['use_shelxe'])
+        self.assertTrue(options.d['shelxe_rebuild'])
+        
+        
+    def test_options_shelxe_resolution_cmd(self):
+        """Check limit with coiled-coil mode turned on
+        """
+        extra = (REBUILD_MAX_PERMITTED_RESOLUTION - SHELXE_MAX_PERMITTED_RESOLUTION_CC) / 2
+        limit = REBUILD_MAX_PERMITTED_RESOLUTION - extra
```

Comment:
The limits selected could be a little more explicit. I had to double check what exactly was being done here. 

You could also do this using random. For example:

`from random import uniform

limit = uniform(SHELXE_MAX_PERMITTED_RESOLUTION, SHELXE_MAX_PERMITTED_RESOLUTION_CC)
`

and

`limit = uniform(SHELXE_MAX_PERMITTED_RESOLUTION_CC , REBUILD_MAX_PERMITTED_RESOLUTION)`

Reply:
Sigh. I think I preferred it when you didn't know what you were talking about...

Very good points. Tests updated accordingly.